community/kernel/src/main/java/org/neo4j/unsafe/impl/batchimport/staging/OnDemandDetailsExecutionMonitor.java

```diff
@@ -0,0 +1,279 @@
+/*
+ * Copyright (c) 2002-2017 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.unsafe.impl.batchimport.staging;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.PrintStream;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.LongFunction;
+
+import org.neo4j.graphdb.DependencyResolver;
+import org.neo4j.helpers.Format;
+import org.neo4j.helpers.collection.Pair;
+import org.neo4j.kernel.impl.cache.MeasureDoNothing;
+import org.neo4j.kernel.impl.cache.MeasureDoNothing.CollectingMonitor;
+import org.neo4j.kernel.impl.util.OsBeanUtil;
+import org.neo4j.unsafe.impl.batchimport.stats.DetailLevel;
+import org.neo4j.unsafe.impl.batchimport.stats.Keys;
+import org.neo4j.unsafe.impl.batchimport.stats.Stat;
+import org.neo4j.unsafe.impl.batchimport.stats.StepStats;
+
+import static java.lang.Long.max;
+import static java.lang.System.currentTimeMillis;
+
+import static org.neo4j.helpers.Format.bytes;
+import static org.neo4j.helpers.Format.date;
+import static org.neo4j.helpers.Format.duration;
+
+/**
+ * Sits in the background and collect stats about stages that are executing.
+ * Reacts to console input from {@link System#in} and on certain commands print various things.
+ * Commands (complete all with ENTER):
+ * <p>
+ * <table border="1">
+ *   <tr>
+ *     <th>Command</th>
+ *     <th>Description</th>
+ *   </tr>
+ *   <tr>
+ *     <th>i</th>
+ *     <th>Print {@link SpectrumExecutionMonitor compact information} about each executed stage up to this point</th>
+ *   </tr>
+ * </table>
+ */
+public class OnDemandDetailsExecutionMonitor implements ExecutionMonitor
```

Comment:
can't find tests for this guy as well

Reply:
There's a refactoring ahead of this PR in the making, which pulls out the functionality about reacting to user input... which really most of this class is about. Testing could be added to that instead IMHO.

Reply:
ok